admissions/admissions/serializers.py

```diff
@@ -247,33 +273,61 @@ class Meta:
 
 
 class ApplicationCreateUpdateSerializer(serializers.HyperlinkedModelSerializer):
+    group_applications = serializers.SerializerMethodField()
+
+    def get_group_applications(self, obj):
+        qs = getattr(obj, "group_applications_filtered", obj.group_applications)
+        return ShortGroupApplicationSerializer(qs, many=True).data
+
     class Meta:
         model = UserApplication
-        fields = ("text", "pk", "phone_number")
+        fields = ("pk", "responses", "group_applications")
+
+    def _validate(self, group_name: str, questions: dict, responses: dict) -> None:
+        error = self._validate_fields(questions, responses)
+        if error is not None:
+            if group_name == None:
+                raise serializers.ValidationError({"responses": error})
+            raise serializers.ValidationError({"groupResponses": {group_name: error}})
+
+    def _validate_fields(self, questions: dict, responses: dict) -> dict:
+        # Ensure that the received JSON indata matches the requested questions
+        editable_questions = list(filter(lambda x: x["type"] != "text", questions))
+        if len(editable_questions) != len(responses):
+            return "Alle spørsmål ble ikke sendt"
+        for question in editable_questions:
+            print(responses)
+            print(question)
+            response_key = next(
+                (key for key in responses.keys() if key == question["id"]), None
+            )
+            if response_key is None:
+                return {question["id"]: "Obligatorisk felt ble ikke sendt"}
+            response = responses[response_key]
+            if response is None:
+                return {question["id"]: "Obligatorisk felt"}
```

Comment:
Hmm, this looks like it could be quite slow? Not a huge deal here though. Not exactly performance critical software :grimacing:

Reply:
Hmm yeah.. Do you have any ideas from the top of your head about how to improve it? Could be nice to improve, but yeah it's not very ciritical and it's run very few times per application